onnxruntime/test/ir/flatbuffer_utils_test.cc

```diff
@@ -0,0 +1,409 @@
+
+  return initializers;
+}
+#endif // ENABLE_TRAINING_APIS
+
+template <typename T>
+std::vector<T> ConvertRawDataToTypedVector(ONNX_NAMESPACE::TensorProto initializer) {
+  std::vector<T> data;
+  data.resize(initializer.raw_data().size() / sizeof(T));
+  memcpy(data.data(), initializer.raw_data().data(), initializer.raw_data().size());
+  return data;
+}
+
+#define ASSERT_EQ_FB_TENSORPROTO_VECTORFIELD(EXPECTED, ACTUAL, FIELD) \
+  ASSERT_EQ(EXPECTED.FIELD.size(), ACTUAL.FIELD.size());              \
+  for (int j = 0; j < EXPECTED.FIELD.size(); ++j) {                   \
+    ASSERT_EQ(EXPECTED.FIELD[j], ACTUAL.FIELD[j]);                    \
+  }
+
+}  // namespace
+
+// tests method that loads to tensorproto protobuf (used when loading a checkpoint into an inference model)
+TEST(GraphUtilsTest, ExternalWriteReadWithLoadInitializers) {
+  // create data
+  auto initializers = CreateInitializers();
+
+  flatbuffers::FlatBufferBuilder builder(1024);
+
+  // write
+  std::ofstream output_stream;
+  ExternalDataWriter writer;
+  CreateWriter("ExternalWriteReadBasicTest.bin", output_stream, writer);
+
+  std::vector<flatbuffers::Offset<fbs::Tensor>> fbs_tensors;
+  for (const auto& initializer : initializers) {
+    flatbuffers::Offset<fbs::Tensor> fbs_tensor;
+    ASSERT_STATUS_OK(SaveInitializerOrtFormat(builder, initializer, Path(), fbs_tensor, writer));
+    fbs_tensors.push_back(fbs_tensor);
+  }
+
+  // TODO: might be 844 depending on whether it's 4 byte or 8 byte alignment
+  ASSERT_EQ(output_stream.tellp(), 840) << "Data written to the external file is incorrect.";
+  output_stream.close();
+  ASSERT_TRUE(output_stream.good()) << "Failed to close data file.";
+
+  auto fbs_tensors_offset = builder.CreateVector(fbs_tensors);
+  fbs::test::TestDataBuilder tdb(builder);
+  tdb.add_initializers(fbs_tensors_offset);
+  builder.Finish(tdb.Finish());
+  auto fb_data = builder.GetBufferSpan();
+
+  auto test_data = fbs::test::GetTestData(fb_data.data());
+  auto fbs_tensors2 = test_data->initializers();
+
+  // read
+  std::ifstream input_stream;
+  ExternalDataReader reader;
+  CreateReader("ExternalWriteReadBasicTest.bin", input_stream, reader);
+
+  std::vector<ONNX_NAMESPACE::TensorProto> loaded_initializers;
+  OrtFormatLoadOptions options;
+
+  for (const auto* fbs_tensor : *fbs_tensors2) {
+    ONNX_NAMESPACE::TensorProto initializer;
+    ASSERT_STATUS_OK(LoadInitializerOrtFormat(*fbs_tensor, initializer, options, reader));
+    loaded_initializers.emplace_back(std::move(initializer));
+    // also check that the loaded flatbuffer tensors have accurately written to the external_data_offset field
+    if (fbs_tensor->data_type() != fbs::TensorDataType::STRING && fbs_tensor->name()->str() != "tensor_32_small")
+    {
+      ASSERT_TRUE(fbs_tensor->external_data_offset() >= 0) << "external_data_offset is not set when we expect it to be set for tensor " << fbs_tensor->name()->str();
```

Comment:
general: prefer the more specific ASSERT macros as those can show more information about a failure. e.g., ASSERT_GE here